src/main.rs

```diff
@@ -340,84 +340,86 @@ impl Environment {
             };
         }
 
-        let entry_point_addr = *executable.exported_symbols.get("start").ok_or_else(|| {
-            "Mach-O file has no 'start' symbol, perhaps it is not an executable?".to_string()
-        })?;
-        let entry_point_addr = abi::GuestFunction::from_addr_with_thumb_bit(entry_point_addr);
+        if let mach_object::ThreadState::Arm { __pc, .. } = executable.thread_state {
```

Comment:
Can you use `let … else {` so we don't need to indent the rest of the function?

Reply:
We should probably also check the other values in the initial state conform to expectations, in case the app is actually relying on them.

Reply:
The other values that ThreadState has are seemingly important.
![image](https://user-images.githubusercontent.com/57239451/218265473-2bb0c22f-362c-4511-9b8e-9b84d6da0ba1.png)
But I have not seen those used by an app, as they're all zeros except __pc.

Reply:
In that case we can just have an assert something like this:

```
assert!(&_r == &[0; 13] && __sp == 0 && __lr == 0 && __cpsr == 0);
```